nb2/service/person_service.py

```diff
@@ -0,0 +1,58 @@
+from nb2 import db
+from nb2.models import Person
+from nb2.service.dtos import CreatePersonDTO
+from nb2.service.exceptions import EmptyRequiredFieldException
+
+
+def get_all_people():
+    """
+    Return a list of all Person objects in the database.
+
+    Returns:
+        List of Person objects in the database.
+        Empty list if no People in the database.
+    """
+    return Person.query.all()
+
+
+def get_person_by_slack_user_id(slack_user_id: str):
+    """
+    Return Person with slack_user_id if it exists.
+
+    Required Args:
+        slack_user_id: string representing the Person's primary Slack id.
+
+    Returns:
+        Person object if one exists in the db with slack_user_id
+        None if no such person exists.
+    """
+    return Person.query.filter_by(slack_user_id=slack_user_id).one_or_none()
+
+
+def create_person(data: CreatePersonDTO):
+    """
+    Add a new Person object to the database.
+
+    Required args:
+        data: An AddPersonDTO instance.
+
+    Returns:
+        Newly created Person object on success.
```

Comment:
Not sure if you think adding info about what this method raises is worthwhile. I believe it might be possible for it to raise an `IntegrityError` if a person with the `slack_user_id` exists. I'm not sure if there are others...

Reply:
I guess also it raises the error that is explicitly raised in the body `EmptyRequiredFieldException`

Reply:
Good catch. Yeah it definitely could return an `IntegrityError` - I'll add that in